application-task-ui/src/main/resources/TaskManager/TaskPagesSearch.xml

```diff
@@ -0,0 +1,186 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="TaskManager.TaskPagesSearch" locale="">
+  <web>TaskManager</web>
+  <name>TaskPagesSearch</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>TaskPagesSearch</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#if($xcontext.action == 'get' &amp;&amp; "$!{request.outputSyntax}" == 'plain')
+  $response.setContentType('application/json')
+  #set ($statement = "SELECT distinct doc.fullName, doc.name, doc.space FROM Document doc, doc.object(TaskManager.TaskManagerClass) AS taskObject WHERE doc.fullName &lt;&gt; 'TaskManager.TaskManagerTemplate' AND lower(doc.fullName) LIKE lower(:param)")
+  #set ($param = $services.query.parameter().anyChars().literal("$!request.text").anyChars())
+  #set ($query = $services.query.xwql($statement).setLimit(10).bindValue('param', $param))
+  #set ($array = [])
+  #foreach ($item in $query.execute())
+    #set ($docRef = $services.model.resolveDocument($item[0]))
+    #if ($services.security.authorization.hasAccess("view", $docRef))
+      #set ($discard = $array.add({
+        'label': $item[1],
+        'value': $item[0],
+        'hint':  $item[2]
+      }))
+    #end
+  #end
+  $jsontool.serialize($array)
```

Comment:
When you output the JSON like this you risk triggering rendering transformations (like icon transformation, e.g. if you have ``:)`` somewhere in the JSON). The safe way to produce JSON from a wiki page is to use the ``#jsonResponse`` Velocity macro, see https://github.com/xwiki/xwiki-platform/blob/e33896a32bad518f13889eceff7fe2859e391c55/xwiki-platform-core/xwiki-platform-xclass/xwiki-platform-xclass-ui/src/main/resources/XWiki/XWikiClassesLiveTableResults.xml#L55 .
* no need to set the response content type yourself
* you can set ``output="false"`` on the Velocity macro, or at least ``wiki="false"``